protocol/0087-VAMM-automated_market_maker.md

```diff
@@ -0,0 +1,94 @@
+# Automated Market Maker Framework
+
+## Summary
+
+The automated market maker (AMM) framework is designed to allow for the provision of an on-chain market making methodology which automatically provides prices according to a simple set of rules based on current market data. These rulesets are not created with the expectation of providing any profit nor of remaining solvent under any specific conditions, and so should be limited to conceptually simple setups. The initial methodology follows a concentrated-liquidity style constant-function market setup, with configurable maximum and minimum price bounds.
+
+An automated market maker is configured at a per-key level, and is enabled by submitting a transaction with the requisite parameters. At this point in time the protocol will move committed funds to a sub-account which will be used to manage order and position margin for the AMM. Once enabled, the configuration will be queried once per block and the resultant orders will be placed on the book, combined with all current orders from the key being cancelled. 
+
+Each party may have only one AMM configuration per market.
+
+## Process Overview
+
+The configuration and resultant lifecycle of an automated market maker is as follows:
+
+- Party funds a key which will be used by the strategy with desired token amounts.
+- Party submits a transaction containing configuration for the strategy on a given market. This will contain:
+  - Amount of funds to commit
+  - Price bounds (upper, lower, base)
+  - Granularity of price levels to post
```

Comment:
This is only relevant if we put volume directly on book, no? 

If we go via the route @barnabee was describing we'd register each source range with the order book and then when a price level is empty on the order book ask the AMM: how much volume at this price?

Reply:
Yes, some more of this spec needs to be killed/reworded if we end up in the off-book AMM world